src/Plugins/ComplexCore/test/RawBinaryReaderTest.cpp

```diff
@@ -0,0 +1,148 @@
+  DataStructure ds;
+  Arguments args;
+
+  // Create default Parameters for the filter.
+  args.insertOrAssign(RawBinaryReaderFilter::k_InputFile_Key, std::make_any<FileSystemPathParameter::ValueType>(fs::path(inputFile)));
+  args.insertOrAssign(RawBinaryReaderFilter::k_ScalarType_Key, std::make_any<NumericType>(NumericType::uint8));
+  args.insertOrAssign(RawBinaryReaderFilter::k_NumberOfComponents_Key, std::make_any<int32>(1));
+  args.insertOrAssign(RawBinaryReaderFilter::k_Endian_Key, std::make_any<ChoicesParameter::ValueType>(0));
+  args.insertOrAssign(RawBinaryReaderFilter::k_SkipHeaderBytes_Key, std::make_any<uint64>(0));
+  args.insertOrAssign(RawBinaryReaderFilter::k_CreatedAttributeArrayPath_Key, std::make_any<DataPath>(createdAttributeArrayPathValue));
+
+  // Preflight the filter and check result
+  auto preflightResult = filter.preflight(ds, args);
+  REQUIRE(preflightResult.outputActions.valid());
+
+  // Execute the filter and check the result
+  auto executeResult = filter.execute(ds, args);
+  REQUIRE(executeResult.result.valid());
+
+  // Check the results
+  auto* resultArray = ds.getDataAs<DataArray<uint8>>(createdAttributeArrayPathValue);
+  for(int i = 0; i < 10; i++)
+  {
+    uint8 resultVal = resultArray->at(i);
+    REQUIRE(resultVal == i);
+  }
+}
+
+TEST_CASE("Core::RawBinaryReaderFilter: Invalid filter execution")
+{
+  std::string arrayName = "Imported Array";
+  DataPath createdAttributeArrayPathValue = DataPath(std::vector<std::string>{arrayName});
+
+  // Instantiate the filter, a DataStructure object and an Arguments Object
+  RawBinaryReaderFilter filter;
+  DataStructure ds;
+  Arguments args;
+
+  // Test empty input file
+  std::string inputFile;
+
+  args.insertOrAssign(RawBinaryReaderFilter::k_InputFile_Key, std::make_any<FileSystemPathParameter::ValueType>(fs::path(inputFile)));
+  args.insertOrAssign(RawBinaryReaderFilter::k_ScalarType_Key, std::make_any<NumericType>(NumericType::uint8));
+  args.insertOrAssign(RawBinaryReaderFilter::k_NumberOfComponents_Key, std::make_any<int32>(1));
+  args.insertOrAssign(RawBinaryReaderFilter::k_Endian_Key, std::make_any<ChoicesParameter::ValueType>(0));
+  args.insertOrAssign(RawBinaryReaderFilter::k_SkipHeaderBytes_Key, std::make_any<uint64>(0));
+  args.insertOrAssign(RawBinaryReaderFilter::k_CreatedAttributeArrayPath_Key, std::make_any<DataPath>(createdAttributeArrayPathValue));
+
+  auto preflightResult = filter.preflight(ds, args);
+  REQUIRE(preflightResult.outputActions.invalid());
+
+  // Test non-existent input file
+  inputFile = "/tmp/this_does_not_exist.bin";
+  args.insertOrAssign(RawBinaryReaderFilter::k_InputFile_Key, std::make_any<FileSystemPathParameter::ValueType>(fs::path(inputFile)));
+  preflightResult = filter.preflight(ds, args);
+  REQUIRE(preflightResult.outputActions.invalid());
+
+  // Update InputFile to the correct path and write the file so it exists
+  inputFile = (fs::temp_directory_path() /= "raw_binary_reader_data.bin").string();
```

Comment:
Same as above about temp directories. Be sure to delete the file from the filesystem when the test is done with the file.